### PR TITLE
Add a couple of tools to help with debugging integration tests

### PR DESCRIPTION
**Pull request by ilyagr**
These are a couple of tools I made to help with debugging #864. I was only partially successful, but I think they can be useful regardless.

### CODE REVIEW

**Review by martinvonz — COMMENTED** [review]
> Before this, it was difficult to run an integration test after adding any directives
> from printf-style debugging to jj (e.g. `err!`, `eprintln!`, `println!`), since
> `jj_cmd_success` fails if `jj` to output anything to stderr while `jj_cmd_failure`
> fails if stdout is not empty.

FYI, there's also `jj_cmd` (without `_success` or `_failure`) which doesn't check anything

**Comment by martinvonz** [comment]
> > Before this, it was difficult to run an integration test after adding any directives
> > from printf-style debugging to jj (e.g. `err!`, `eprintln!`, `println!`), since
> > `jj_cmd_success` fails if `jj` to output anything to stderr while `jj_cmd_failure`
> > fails if stdout is not empty.
> 
> FYI, there's also `jj_cmd` (without `_success` or `_failure`) which doesn't check anything

I now realize that you didn't want to test the output. Sorry :)

**Comment by ilyagr** [comment]
> FYI, there's also jj_cmd (without _success or _failure) which doesn't check anything

Yes, before this PR, the way to do printf-style debugging would be to change each `jj_cmd_success` to `jj_cmd` and then extract stdout/stderr from the result. I did it once; it's not fun.

**Review by martinvonz — APPROVED** [review]

**Comment by ilyagr** [comment]
As a complement to this, I'm thinking of creating a `TestEnvironment::preserve_repo_for_each_jj_invocation` that preserves the repo state before each `jj_cmd` command in a separate directory. This would allow debugging with gdb/lldb at that exact repo state.

(Alternatively, we could only preserve the last state and rely on the operation log, but that would be annoying when testing behavior that does not exist for `jj --at-op`)

I'm not 100% sure that would be a worthwhile endeavor; perhaps I will instead just get used to recreating the desired repo state myself.

File: tests/common/mod.rs [hunk 1]
```diff
@@ -223,6 +237,23 @@ impl TestEnvironment {
...
+    ///
+    /// This would run all the tests that contain `fetch_del` in their name in a
+    /// file that contains `test_git_colocated` and show the output.
+    ///
+    /// This is meant only for debugging; the call to this function should be
+    /// removed when merging to the main branch
+    pub fn allow_eprintln_debugging(&mut self) {
+        self.debug_allow_eprintln = true;
+    }
+
```
[the remark this answers is not in the document]
Comment (ilyagr) [reply tests/common/mod.rs:256]: That sounds like a good idea. My one reservation is that I'm not sure where to put the documentation that I put before the function here. One possibility is to keep the function as-is and also support the environment variable. Another possibility is to create a "debugging.md" file or add a "Debugging" section to contributing.md.

[the remark this answers is not in the document]
Comment (martinvonz) [reply tests/common/mod.rs:256]: I doubt I would have looked for the method here if I were trying to get printf debugging to work in tests. I think it mostly helps developers who happen to see the documentation here. So my guess is that it won't be much worse to just leave this comment above wherever we end up reading the environment variable.

[the remark this answers is not in the document]
Comment (ilyagr) [reply tests/common/mod.rs:256]: I'm now thinking of making `TestEnvironment::debug_allow_eprintln` boolean public, documenting it with this docstring, and defaulting it to true when an environment variable is set.

**Pull request merged**